### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by ihackcode**
Some fixes, might need more testing.

With archive.php and item.php I've tested it with Timezone offsets -4, -5, +3.5 and +4.5. This was the only way I could make the data save consistently. Also, it was the only way I was able to get timezones like +3.5 and +4.5 to work. timezone_name_from_abbr and DateTimezone didn't accept the half hour timezones.

### CODE REVIEW

File: thumb.php [hunk 1]
```diff
@@ -770,7 +770,7 @@ protected function processImageAndWriteToCache($localImage)
         }
 
         // create a new true color image
-        $canvas = imagecreatetruecolor($newWidth, $newHeight);
+        $canvas = imagecreatetruecolor(intval($newWidth), intval($newHeight));
```
Comment (mambax7) [inline thumb.php:773]: For consistency, it would better to use only (int) instead of intval.
Plus (int) is faster :)

File: archive.php [hunk 2]
```diff
@@ -168,8 +168,12 @@
 	
     // must adjust the selected time to server timestamp
     $timeoffset = $useroffset - $GLOBALS['xoopsConfig']['server_TZ'];
-    $monthstart = mktime(0 - $timeoffset, 0, 0, $frommonth, 1, $fromyear);
-    $monthend   = mktime(23 - $timeoffset, 59, 59, $frommonth + 1, 0, $fromyear);
+    $timeoffsethours = intval($timeoffset);
```
Comment (mambax7) [inline archive.php:171]: (int) instead of intval

File: class/Form/ItemForm.php [hunk 3]
```diff
@@ -359,6 +359,7 @@ public function createElements($obj)
             }
 
             $dateExpireYesNo     = new \XoopsFormRadioYN('', 'use_expire_yn', $dateexpire_opt);
+            $dateexpire = \strtotime(\formatTimestamp($dateexpire)); //set to user timezone
```
Comment (mambax7) [inline class/Form/ItemForm.php:362]: Based on the recent discussion: https://xoops.org/modules/newbb/viewtopic.php?start=0&topic_id=78783, we have to get away from using "stringtotime()"
See: https://docs.google.com/document/d/1FEoyDfI96XZn-Ib0Ywigi3nYyj6wkbBagFgbtul0qeA/
Let me know what you think...

**Comment by mambax7** [comment]
Thank you so much!!!

File: class/Form/ItemForm.php (hunk 3 above)
Comment (ihackcode) [reply class/Form/ItemForm.php:362]: thanks. I changed this with my PR #137

**Pull request merged**